### PR TITLE
Add logging instructions in a separate document

### PR DESCRIPTION
**Pull request by sameerag**
We are frequently sharing logging instructions over all help channels including github. Adding a separate doc to make the reference easy as part of github response.

### CODE REVIEW

File: lib/msal-browser/docs/logging.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# MSAL JS logging instructions
+
+These are steps needed to enable MSAL JS logging for your applications:
+
+1. In the MSAL [Configuration](./configuration.md) object, you can enable logging to collect msal js logs. We enable different levels of logging and an appropriate level can be chosen as needed. 
+
+2. The logger options can be set as below, the example chooses `loglevel.trace`
```
Comment (derisen) [inline lib/msal-browser/docs/logging.md]:
```suggestion
2. The logger options can be set as below, the example chooses `LogLevel.Trace`
```

File: lib/msal-browser/docs/logging.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            }
+        }
+    }
+
+```
+
+An example usage in a sample can be accessed [here](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/samples/msal-browser-samples/VanillaJSTestApp2.0/app/default/authConfig.js#:~:text=logLevel%3A%20msal.LogLevel.Trace%2C).
+
+
+3. Make sure you have the "Verbose" level enabled in your browser console to see these logs
```
Comment (derisen) [inline lib/msal-browser/docs/logging.md]: Not sure if this is necessary. Perhaps we can say something like "we recommend enabling verbose level logs in your browser console settings" etc..

**Comment by derisen** [comment]
Should we duplicate this for msal-node as well? (or perhaps keep this under msal-common to share)

File: lib/msal-browser/docs/logging.md (hunk 2 above)
Comment (sameerag) [reply lib/msal-browser/docs/logging.md]: Yes, I would like to keep this as a suggestion since many users forget it. Let me rephrase.

File: lib/msal-browser/docs/logging.md (hunk 1 above)
Comment (sameerag) [reply lib/msal-browser/docs/logging.md]: Done.

File: lib/msal-browser/docs/logging.md (hunk 2 above)
Comment (sameerag) [reply lib/msal-browser/docs/logging.md]: Updated with an illustration and made it more generic.

File: lib/msal-browser/docs/logging.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+2. The logger options can be set as below, the example chooses `LogLevel.Trace`
+
+```javascript
+
+    ...
+    ...
+
+        system: {
+        loggerOptions: {
+            logLevel: msal.LogLevel.Trace,
```
Comment (tnorling) [inline lib/msal-browser/docs/logging.md]: nit: Most people should be consuming the lib from npm where we don't use the `msal` namespace. Can we remove it from the example here and show the proper imports at the top?

File: lib/msal-browser/docs/logging.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+These are steps needed to enable MSAL JS logging for your applications:
+
+1. In the MSAL [Configuration](./configuration.md) object, you can enable logging to collect msal js logs. We enable different levels of logging and an appropriate level can be chosen as needed. 
+
+2. The logger options can be set as below, the example chooses `LogLevel.Trace`
+
+```javascript
+
+    ...
```
Comment (tnorling) [inline lib/msal-browser/docs/logging.md]: nit: For completeness we should show that this is the object that gets passed into `PublicClientApplication`

**Review by tnorling — COMMENTED** [review]
This is long overdue, thanks for putting this together! Just a few nits but otherwise looks good

File: lib/msal-browser/docs/logging.md (hunk 4 above)
Comment (derisen) [reply lib/msal-browser/docs/logging.md]: +1

Comment (sameerag) [reply lib/msal-browser/docs/logging.md]: Done

File: lib/msal-browser/docs/logging.md (hunk 3 above)
Comment (sameerag) [reply lib/msal-browser/docs/logging.md]: @derisen
Can you address this?

**Comment by sameerag** [comment]
> Should we duplicate this for msal-node as well? (or perhaps keep this under msal-common to share)

Yes. I want this in browser as this speaks of browser specific debug. We can write one for node and add the platform context. I can put some together, if you have a preference for a specific flow in node, let me know.

File: lib/msal-browser/docs/logging.md (hunk 3 above)
Comment (derisen) [reply lib/msal-browser/docs/logging.md]: Yeah probably the better approach for clarity and as default. Folks who are not consuming it from npm shouldn't have much trouble figuring it out

File: lib/msal-browser/docs/logging.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
+# MSAL JS logging instructions
+
+These are steps needed to enable MSAL JS logging for your applications:
+
+1. In the MSAL [Configuration](./configuration.md) object, you can enable logging to collect msal js logs. We enable different levels of logging and an appropriate level can be chosen as needed. 
+
+2. The logger options can be set as below, the example chooses `LogLevel.Trace`
+
+```javascript
+
```
Comment (derisen) [inline lib/msal-browser/docs/logging.md:10]:
```suggestion
import { PublicClientApplication, LogLevel } from "@azure/msal-browser";
```

**Review by derisen — APPROVED** [review]

File: lib/msal-browser/docs/logging.md (hunk 5 above)
Comment (sameerag) [reply lib/msal-browser/docs/logging.md:10]: Ah! I meant in the actual sample. This can be added here, yes.

Comment (derisen) [reply lib/msal-browser/docs/logging.md:10]: Ah ok, sorry. I think we have both variety in the samples (vanilla js samples uses the msal namespace, while react and angular samples import things ES6 style)

File: lib/msal-browser/docs/logging.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    },
+    system: {
+        loggerOptions: {
+            logLevel: LogLevel.Trace,
+            loggerCallback: (level, message, containsPii) => {
+                if (containsPii) {	
+                    return;	
+                }
+                switch (level) {	
+                    case msal.LogLevel.Error:
```
Comment (tnorling) [inline lib/msal-browser/docs/logging.md]:
```suggestion
                    case LogLevel.Error:	
```

File: lib/msal-browser/docs/logging.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+            logLevel: LogLevel.Trace,
+            loggerCallback: (level, message, containsPii) => {
+                if (containsPii) {	
+                    return;	
+                }
+                switch (level) {	
+                    case msal.LogLevel.Error:	
+                        console.error(message);	
+                        return;	
+                    case msal.LogLevel.Info:
```
Comment (tnorling) [inline lib/msal-browser/docs/logging.md]:
```suggestion
                    case LogLevel.Info:	
```

File: lib/msal-browser/docs/logging.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+                    return;	
+                }
+                switch (level) {	
+                    case msal.LogLevel.Error:	
+                        console.error(message);	
+                        return;	
+                    case msal.LogLevel.Info:	
+                        console.info(message);	
+                        return;	
+                    case msal.LogLevel.Verbose:
```
Comment (tnorling) [inline lib/msal-browser/docs/logging.md]:
```suggestion
                    case LogLevel.Verbose:	
```

File: lib/msal-browser/docs/logging.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+                    case msal.LogLevel.Error:	
+                        console.error(message);	
+                        return;	
+                    case msal.LogLevel.Info:	
+                        console.info(message);	
+                        return;	
+                    case msal.LogLevel.Verbose:	
+                        console.debug(message);	
+                        return;	
+                    case msal.LogLevel.Warning:
```
Comment (tnorling) [inline lib/msal-browser/docs/logging.md]:
```suggestion
                    case LogLevel.Warning:	
```

File: lib/msal-browser/docs/logging.md (hunk 6 above)
Comment (sameerag) [reply lib/msal-browser/docs/logging.md]: Done

File: lib/msal-browser/docs/logging.md (hunk 7 above)
Comment (sameerag) [reply lib/msal-browser/docs/logging.md]: Done

File: lib/msal-browser/docs/logging.md (hunk 8 above)
Comment (sameerag) [reply lib/msal-browser/docs/logging.md]: Done

File: lib/msal-browser/docs/logging.md (hunk 9 above)
Comment (sameerag) [reply lib/msal-browser/docs/logging.md]: Done

**Review by tnorling — APPROVED** [review]
Looks like the beachball check isn't ignoring .png files so you'll likely need to admin merge FYI

**Comment by sameerag** [comment]
> Looks like the beachball check isn't ignoring .png files so you'll likely need to admin merge FYI

Yep. Thanks for the headsup. Should we add a pattern in `beachballrc` to ignore `png` files in `docs` folders? I want to add more pngs anyway.
